### PR TITLE
switch to geometrybasics

### PR DESCRIPTION
**Pull request by SimonDanisch**
Needs:
- [x] https://github.com/JuliaGeometry/GeometryBasics.jl/pull/18
- [x] https://github.com/JuliaPlots/MakieGallery.jl/pull/245
- [x] https://github.com/JuliaGeometry/Packing.jl/pull/6
- [x] https://github.com/JuliaGraphics/FreeTypeAbstraction.jl/pull/35
- [x] https://github.com/SimonDanisch/ShaderAbstractions.jl/pull/9

- [x] https://github.com/JuliaIO/MeshIO.jl/pull/56
- [ ] https://github.com/JuliaPlots/GLMakie.jl/pull/76
- [ ] https://github.com/JuliaPlots/WGLMakie.jl/pull/27
- [ ] https://github.com/JuliaPlots/StatsMakie.jl/pull/120
- [ ] https://github.com/jkrumbiegel/GridLayoutBase.jl/pull/1

### CODE REVIEW

File: src/interfaces.jl [hunk 1]
```diff
@@ -90,21 +88,13 @@ $(ATTRIBUTES)
 @recipe(Volume, x, y, z, volume) do scene
     Theme(;
         default_theme(scene)...,
-        fxaa = true,
-        algorithm = :iso,
-        absorption = 1f0,
-        isovalue = 0.5f0,
-        isorange = 0.05f0,
+        algorithm = :mpi,
```
Comment (asinghvi17) [inline src/interfaces.jl]: Did you mean :mip?

File: src/colorsampler.jl [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    apply_scaling(value::Number, scaling::Scaling)
+
+Scales a number to the range 0..1.
+"""
+function apply_scaling(value::Number, scaling::NoScaling)::Float64
+    return Float64(value)
+end
+
+function apply_scaling(value::Number, scaling::Scaling)::Float64
+    value_scaled = scaling.scaling_function(value)
```
Comment (asinghvi17) [inline src/colorsampler.jl]: This would throw an error if given log(0) - would we need to handle this directly in `Sampler`?

Comment (SimonDanisch) [reply src/colorsampler.jl]: Good point!

File: test/simple.jl [hunk 3]
```diff
@@ -1,11 +1,13 @@
 using AbstractPlotting
 using GLMakie
 using GeometryBasics
+using Observables
+using GLMakie
```
Comment (mschauer) [inline test/simple.jl:5]:
```suggestion
```

**Comment by SimonDanisch** [comment]
All 242 tests run without error now :) 
Remaining problem:
- [ ] Still needs to fix 3d contour volume plots.

**Comment by SimonDanisch** [comment]
Ok fuck it! This circular dependency is just too annoying for CI :D I'll merge & tag, and come back to fix CI...

**Pull request merged**